docs/_source/docs/architecture.rst

```diff
@@ -135,15 +135,222 @@ to demonstrate ways of organising different projects.
+  resources.
+* Developers will need to be able to stand up their own, isolated environments of Indigo that do not
+  conflict with production. These environments will have some different configurations to make them
+  cheaper for the company, but otherwise be identical to production.
+
+Project Structure
+*****************
+
+.. code-block:: text
+
+    config/
+        config.yaml             # top-level project configuration
+        indigo/                 # StackGroup for all Indigo's resources
+            config.yaml         # StackGroup Config for Indigo
+            application.yaml    # Stack Config for Indigo's application stack
+        project/                # StackGroup for resources needed to deploy Indigo
+            deployment.yaml     # Stack Config for Indigo's deployment resources
+    templates/
+        indigo/
+            application.yaml
+        project/
+            deployment.yaml
+    vars/
+        production.yaml         # var-file with the production environment's configurations
+        development.yaml        # var-file with configurations for development environments
+
+Example Configs
+***************
+
+Top-level project configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/config.yaml
+
+    # This is the top-level config file for the whole project. Every stack in the project will
+    # inherit these settings unless they explicitly override them. At minimum, you need the
+    # project_code and region, but you can set whatever StackGroup and Stack configurations you
+    # want inherited by the rest of the project.
+
+    # By making the project-code parameterized by environment name, it guarantees that one
+    # environment's stacks do not conflict with another environment's stacks.
+    project_code: indigo-{{ var.environment }}
+    region: {{ var.region }}
+
+    # Setting the template_bucket_name with a !stack_output means that every stack in the project
+    # (except the deployment stack) will have its template uploaded to that bucket. It ALSO means
+    # that every stack in the project will inherit a dependency on the deployment stack (except the
+    # deployment stack... see its comments below)
+    template_bucket_name: !stack_output project/deployment.yaml::BucketName
+
+    # By making the template_key_prefix parameterized by environment, it means that every
+    # environment's templates will be uploaded to its own directory structure within the template
+    # bucket, preventing conflicts.
+    template_key_prefix: {{ var.environment }}
+
+    # Setting the iam_role on the project like this means that all Sceptre actions on all stacks in
+    # the project will be done by assuming the deployment role whose ARN is output by the deployment
+    # stack (except for actions on the deployment stack... see its comments below).
+    iam_role: !stack_output project/deployment.yaml::DeploymentRoleArn
+
+
+The Deployment Stack
```

Comment:
The order of describing these files is a bit confusing since config/config.yaml depends on config/project/deployment.yaml.  Would it make sense to put the description of the deployment stack above config/config.yaml?

Reply:
I don't know if re-ordering would be easier to understand. There's partially a circular dependency here (broken by the is_project_dependency stack config value). One way or another, we need to "break into the circle". I'll reread through this and try to find a better order/way to explain things.

Basically, here's the scenario I'm trying to indicate:
* All stacks in the project "inherit" values from config/config.yaml. Included in this are the dependencies on the deployment stack.
* The deployment stack (which is in the project) disallows dependencies; Instead, it only _provides_ dependencies for the other stacks in the project.